templates/client.mustache

```diff
@@ -249,18 +249,22 @@ func parameterToJson(obj interface{}) (string, error) {
 // callAPI do the request.
 func (c *APIClient) callAPI(request *http.Request) (*http.Response, error) {
 	if c.cfg.Debug {
-		r := request.Clone(context.TODO())
-		r.Body, _ = request.GetBody()
-		h := r.Header
-		if len(h.Get("X-Auth-Token")) != 0 {
-			h.Set("X-Auth-Token", "**REDACTED**")
+		authToken, hasAuth := request.Header["X-Auth-Token"]
+		if hasAuth {
+			request.Header.Set("X-Auth-Token", "**REDACTED**")
```

Comment:
I believe this change can create race conditions when one request object is being used to make multiple API requests, as in a go func loop, especially where execution timing could impact the request.

Reply:
Here's a ChatGPT generated example where the race is triggered in most cases (an actual remote request should reduce the occurrence)
https://gist.github.com/displague/2bda2d2176e255da968478455d0da96d

Reply:
By redacting a debug-only copy, we previously avoid that case.
We could test the ContentLength of nil'ness of the Body to avoid triggering the nil pointer exception.

https://pkg.go.dev/net/http#Request

Reply:
This SDK makes a single request per request object, and that can't be changed without significant refactoring to the request templates: https://github.com/equinix-labs/metal-go/blob/main/templates/api.mustache#L355-L360

Reply:
I'm mostly concerned from a best practice perspective; I agree the likelihood of hitting the race would be near 0.